### DISCUSSION TITLE
Best Practices for multi directory configuration

### DISCUSSION THREAD

**Discussion by AndreiTapai** [littleK0i/SnowDDL#110] [Q&A]
Wondering what the best practice is for situations in which the SnowDDL yaml configuration files are split into multiple paths.

Lets say we have the following structure:
```
snowddl/
├─ __generated__/
│  ├─ my_database/
│  │  ├─ my_schema/
│  │  │  ├─ params.yaml
│  │  │  ├─ table/
│  │  │  │  ├─ codegen_table_1.yaml
│  │  │  │  ├─ codegen_table_2.yaml
├─ custom/
│  ├─ my_database/
│  │  ├─ my_schema/
│  │  │  ├─ table/
│  │  │  │  ├─ custom_table.yml
```

First of all, if we are using snowddl-singledb is it advisable/possible to structure things in such a way? Would I have to run the snowddl CLI command once for each location or is there a way to feed the full configuration in one go? For code cleanliness and code review purposes I want to split out object configuration that is auto-generated from objects that are hand-coded.

**Comment by littleK0i** [comment]
SingleDB mode has exactly the same config as normal mode. You may run it multiple times with different `--source-db` argument to apply config to multiple databases. But it sounds like normal mode to me.

Splitting config into multiple folders is currently not supported, since it is not clear how to merge and resolve conflicts in case name collisions. But you may create your own wrapper for SnowDDL runs and do copy from each config sub-directory into tmp path. After files from all sub-directories were copied, run SnowDDL normally with merged config.

Alternatively, you may use [programmatic config](https://docs.snowddl.com/advanced/programmatic-config) instead of code-generated files.
